server/utils/validation/auth/signupValidation.ts

```diff
@@ -1,15 +1,16 @@
-import Joi from 'joi';
+import { string, object, ref } from 'yup';
+
 import { SignupBodyInterface } from '../../interfaces/signupBodyInterface';
 
 export const validateSignup = (body: SignupBodyInterface) => {
-  const schema = Joi.object({
-    name: Joi.string().required(),
-    email: Joi.string().email().required(),
-    password: Joi.string().regex(/^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&])[a-zA-Z0-9!@#$%^&]{5,}$/).required(),
-    passwordConfirmation: Joi.ref('password'),
-    location: Joi.string().required(),
-    phone: Joi.string().required(),
+  const schema = object({
+    name: string().required(),
+    email: string().email().required(),
+    password: string().matches(/^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&])[a-zA-Z0-9!@#$%^&]{5,}$/).required(),
+    passwordConfirmation: ref('password'),
+    location: string().required(),
+    phone: string().required(),
   });
 
-  return schema.validateAsync(body);
+  return schema.validate(body);
```

Comment:
the validate api is pretty self sufficient so no need to define a function to wrap it
its redundant
the schema's job is to provide the type safety to you

Reply:
Do you mean that I need to export just the schema and call the validate method from where I need to validate the body?

Reply:
yes